scvi/dataset/pbmc.py

```diff
@@ -1,201 +1,134 @@
-        remove_extracted_data: bool = False,
-        delayed_populating: bool = False,
-    ):
-        self.dataset_names = np.asarray(
-            [
-                "cd4_t_helper",
-                "regulatory_t",
-                "naive_t",
-                "memory_t",
-                "cytotoxic_t",
-                "naive_cytotoxic",
-                "b_cells",
-                "cd4_t_helper",
-                "cd34",
-                "cd56_nk",
-                "cd14_monocytes",
-            ]
-        )
-        subset_datasets = subset_datasets if subset_datasets else slice(None)
-        self.remove_extracted_data = remove_extracted_data
-        self.dataset_names = self.dataset_names[subset_datasets]
-        super().__init__(save_path=save_path, delayed_populating=delayed_populating)
-
-        if delayed_populating:
-            for dataset_name in self.dataset_names:
-                Dataset10X(
-                    dataset_name,
-                    save_path=save_path,
-                    delayed_populating=delayed_populating,
-                )
-
-        self.filter_genes_by_count()
-        self.filter_cells_by_count()
-
-    def populate(self):
-        datasets = []
-        for dataset_name in self.dataset_names:
-            dataset = Dataset10X(
-                dataset_name,
-                save_path=self.save_path,
-                remove_extracted_data=self.remove_extracted_data,
-            )
-            dataset.initialize_mapped_attribute(
-                "labels", "cell_types", np.asarray([dataset_name], dtype="<U128")
-            )
-            datasets += [dataset]
-        self.populate_from_datasets(datasets)
+import anndata
+from scvi.dataset import dataset10X
+
+# from scvi.dataset.dataset10X import Dataset10X
+
+from scvi.dataset._utils import _download
+from scvi.dataset import setup_anndata
+
+
+def purified_pbmc_dataset(
+    save_path: str = "data/", subset_datasets: List[str] = None, run_setup_anndata=True
+):
+    # url = "https://github.com/YosefLab/scVI-data/raw/master/PurifiedPBMCDataset.h5ad"
+    # save_fn = "PurifiedPBMCDataset.h5ad"
+
+    # _download(url, save_path, save_fn)
+    # path_to_file = os.path.join(save_path, save_fn)
+
+    path_to_file = "/Users/galen/scVI/galen/PurifiedPBMCDataset.h5ad"
+    adata = anndata.read(path_to_file)
+
+    dataset_names = [
+        "cd4_t_helper",
+        "regulatory_t",
+        "naive_t",
+        "memory_t",
+        "cytotoxic_t",
+        "naive_cytotoxic",
+        "b_cells",
+        "cd4_t_helper",
+        "cd34",
+        "cd56_nk",
+        "cd14_monocytes",
+    ]
+    if subset_datasets is not None:
+        row_indices = []
+        for dataset in subset_datasets:
+            assert dataset in dataset_names
+            idx = np.where(adata.obs["cell_types"] == dataset)[0]
+            row_indices.append(idx)
+        row_indices = np.concatenate(row_indices)
+        adata = adata[row_indices].copy()
+
+    if run_setup_anndata:
```

Comment:
This is a design question. Either the setup ann data is automatic or it's not. We can't sometimes do it for the user and sometimes not

Reply:
changed to always do it